scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,153 @@
+
+    function GetFiles([string]$folder)
+    {
+        if(Test-Path $folder)
+        {
+            $files = Get-ChildItem $folder -recurse
+            return $files
+        }
+        return $null
+    }
+
+    function GetNuGetFoldersPath()
+    {
+        $nugetFolder = [System.IO.Path]::Combine($env:UserProfile, "np")
+        return $nugetFolder
+    }
+
+    function IsClientDotnetExe([string]$nugetClient)
+    {
+        return $nugetClient.endswith("dotnet.exe")
+    }
+
+    function SetupNuGetFolders([string]$_nugetClientFilePath)
+    {
+        $nugetFolders = GetNuGetFoldersPath
+        $Env:NUGET_PACKAGES = [System.IO.Path]::Combine($nugetFolders, "gpf")
+        $Env:NUGET_HTTP_CACHE_PATH = [System.IO.Path]::Combine($nugetFolders, "hcp")
+        $Env:NUGET_PLUGINS_CACHE_PATH = [System.IO.Path]::Combine($nugetFolders, "pcp")
+
+        if($(IsClientDotnetExe $_nugetClientFilePath))
+        {
+            . $_nugetClientFilePath nuget locals -c all  *>>$null
+        } 
+        else 
+        {
+            . $_nugetClientFilePath locals -clear all -Verbosity quiet
+        }
+    }
+
+    function SetupGitRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    {
+        if(!(Test-Path $sourceDirectoryPath))
+        {
+                git clone $repository $sourceDirectoryPath
+                git -c $sourceDirectoryPath checkout $commitHash
+        }
+        else 
+        {
+                Log "Skipping the cloning of $repository as $sourceDirectoryPath is not empty" -color "Yellow"
+        }
+
+        $gitRepoName = $repository.Substring($($repository.LastIndexOf('/') + 1))
+        $potentialSolutionFile = [System.IO.Path]::Combine($sourceDirectoryPath, "$($gitRepoName.Substring(0, $gitRepoName.Length - 4)).sln")
+
+        if(Test-Path $potentialSolutionFile)
+        {
+            $solutionFile = $potentialSolutionFile
+        } 
+        else 
+        {
+            $solutionFile = (Get-ChildItem $sourceDirectoryPath *.sln)[0] | Select-Object -f 1 | Select-Object -ExpandProperty FullName
+        }
+        Log "Completed the repository setup. The solution file is $solutionFile" -color "Green"
+        return $solutionFile
+    }
+
+    function RunPerformanceTestsOnGitRepository([string]$nugetClient, [string]$sourceRootDirectory, [string]$repoUrl,  [string]$commitHash, [string]$resultsDirPath, [string]$logsPath)
+    {
+        $repoName = GenerateNameFromGitUrl $repoUrl
+        $resultsFilePath = [System.IO.Path]::Combine($resultsDirPath, "$repoName.csv")
+        $solutionFilePath = SetupGitRepository $repoUrl $commitHash $([System.IO.Path]::Combine($sourceRootDirectory, $repoName))
+        SetupNuGetFolders $nugetClient
+        . "$PSScriptRoot\RunPerformanceTests.ps1" $nugetClient $solutionFilePath $resultsFilePath $logsPath
+    }
+    
+
+    # The format of the URL is assumed to be https://github.com/NuGet/NuGet.Client.git. The result would be NuGet-Client-git
```

Comment:
👍 for the comment explaining the method.... Consider adding comments to the rest of them.